datastream-cloud-storage/src/main/java/com/linkedin/datastream/cloud/storage/io/ScribeParquetAvroConverter.java

```diff
@@ -0,0 +1,336 @@
+                  }
+                  record.put(fieldName.toLowerCase(), result);
+                } else {
+                  record.put(fieldName.toLowerCase(), avroRecord.get(fieldName));
+                }
+              } else if (avroTypeName.equalsIgnoreCase("RECORD") && avroTypeSchema.getName().equalsIgnoreCase("scribe_header") && scribeHeaderRecord != null) {
+                record.put(fieldName.toLowerCase(), scribeHeaderRecord.get(fieldName));
+              } else if (avroTypeName.equalsIgnoreCase("RECORD")) {
+                //convert fields inside nested object to parquet data
+                GenericRecord nestedRecord = (GenericRecord) avroRecord.get(fieldName);
+                Schema nestedRecordschema = schema.getField(fieldName.toLowerCase()).schema().isUnion() ? schema.getField(fieldName.toLowerCase()).schema().getTypes().get(1) : schema.getField(fieldName.toLowerCase()).schema();
+                record.put(fieldName.toLowerCase(), generateParquetStructuredAvroData(nestedRecordschema, nestedRecord));
+              } else {
+                // this is for primitive data types which do not need any conversion
+                record.put(fieldName.toLowerCase(), avroRecord.get(fieldName));
+              }
+            } catch (Exception e) {
+              LOG.error(String.format("Exception in converting avro field to parquet in ScribeParquetAvroConverter: Schema: %s, field: %s, typeName: %s, exception: %s", schema.getName(), fieldName, avroTypeName, e));
+            }
+          }
+        }
+      } catch (NullPointerException e){
+        LOG.error(String.format("NPE while converting avro field to parquet in ScribeParquetAvroConverter: Schema: %s, field: %s", schema.getName(), fieldName));
+      } catch (Exception e) {
+        LOG.error(String.format("Error in converting avro field to parquet in ScribeParquetAvroConverter: Schema: %s, field: %s", schema.getName(), fieldName));
+      }
+    }
+    return record;
+  }
+
+  /**
+   * Get Parquet formatted scribe header data
+   * @param avroRecord
+   * @return Parquet-format record
+   */
+  public static GenericRecord getScribeHeaderParquetData(GenericRecord avroRecord, Schema parquetSchema) {
+    Schema scribeHeaderSchema = avroRecord.getSchema().getField(SCRIBE_HEADER).schema();
+    Schema scribeHeaderTypeSchema = scribeHeaderSchema.isUnion() ? scribeHeaderSchema.getTypes().get(1) : scribeHeaderSchema;
+    return generateParquetStructuredAvroData(generateParquetStructuredAvroSchema(scribeHeaderTypeSchema), (GenericRecord) avroRecord.get(SCRIBE_HEADER));
+  }
+
+  /**
+   * Convert long epoch timestamp value to String date time format
+   * @param value long epoch timestamp
+   * @return String datetime format
+   */
+  private static String convertDateTimeToString(Long value) {
+    DateFormat format = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss.SSS ZZ");
+    String date = format.format(new Date(value));
```

Comment:
You should set it to EST explicitly since GCP is UTC by default.

Reply:
As discussed offline, this needed further investigation. [Here](https://projecthub.service.csnzoo.com/browse/ACISC-3552) is the ticket for it